apis/management.cattle.io/v3/authz_types.go

```diff
@@ -116,3 +124,7 @@ type ClusterRoleTemplateBinding struct {
 	ClusterName        string `json:"clusterName,omitempty" norman:"required,type=reference[cluster]"`
 	RoleTemplateName   string `json:"roleTemplateName,omitempty" norman:"required,type=reference[roleTemplate]"`
 }
+
+type SetPodSecurityPolicyTemplateInput struct {
+	PodSecurityPolicyTemplate string `json:"podSecurityPolicyTemplate" norman:"required"`
```

Comment:
follow saming naming paradign as above

Reply:
regarding this comment:
> follow saming naming paradign as above
Note that once you have it setup properly, the UI should display a drop down of pspts, not a text box
![psptaction](https://user-images.githubusercontent.com/2473240/37666346-9664abfe-2c1c-11e8-917d-edd0118254ff.png)

Reply:
and testing this has led be to discover that the PSPT is misconfigured. It should have a displayName field and the displayName mapper.
Right now, when i create one, the name given to it is used as the ID. This is incorrect. The id should be autogenerated and their shoudl be a separate API name field with the name the user entered. This api name field should be stored in K8S as displayName (to avoid naming collision with objectmeta.Name, which k8s uses as the id).

When fixed, the url for a pspt should looke like this:
```
https://localhost:8443/v3/podSecurityPolicyTemplates/<semi random thing that will look like podsecuritypolicytempalte-kq3jh>
```
not this (which is what it is currently):
```
https://localhost:8443/v3/podSecurityPolicyTemplates/test1
```